boards/arduino/opta/arduino_opta_stm32h747xx_m4_defconfig

```diff
@@ -13,7 +13,7 @@ CONFIG_ARM_MPU=y
 # Enable HW stack protection
 CONFIG_HW_STACK_PROTECTION=y
 
-# enable pin controller
+# Enable pin control
```

Comment:
this must not be done at board level, drivers must do this

Reply:
Sorry, what exactly should not be done? If you mean enabling pin control, I just copied this from other STM32 boards, see for example: `zephyr/boards/st/stm32h747i_disco/stm32h747i_disco_stm32h747xx_m7_defconfig`. The same is true for `CONFIG_HW_STACK_PROTECTION`.

Reply:
pinctrl

Reply:
But if I remove it even the simplest sample (blinky) fails because of missing symbols:
```
/home/fog/Workspaces/zephyrproject/zephyr/include/zephyr/drivers/pinctrl.h:350: undefined reference to `pinctrl_lookup_state'
```
I checked 10 random boards in `zephyr/boards/st` and all have `CONFIG_PINCTRL=y` in defconfig. Given that the Opta is basically an stm32h747i why can't it be configured the same way?

Reply:
Because that is wrong, @gmarull can explain

Reply:
Also @erwango you need to fix your drivers

Reply:
@fogzot To add some context, IIUC, the proper solution would be to have this symbol automatically `select`ed via Kconfig by drivers that need it, like this USB driver for some NXP device does:  

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/usb/device/Kconfig#L166-L173

The STM32 USB driver does not seem to do that:

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/usb/device/Kconfig#L46-L56

... even though it uses it internally, e.g.:

https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/usb/device/usb_dc_stm32.c#L444

This is an issue with some current Zephyr drivers that should be fixed treewide. Personally I don't see this as a reason to block a PR adding support for additional cores, but that's just my opinion.

Reply:
I understand the problem but if I apply the requested change my pull request will never pass the automatic checks because the tests won't build. Sincerely I don't like the idea of putting this on hold until the drivers are fixed but I don't see how to proceed otherwise. Suggestions very welcome.

Reply:
> Also @erwango you need to fix your drivers

@nordicjm 

On the form:
This feels as an aggressive statement. Please rephrase.

On the content:
Ok, sure this can be done. And I agree it makes sense, but it isn't a but per se and I think you'll agree this is not linked to this board, so no reason to block.
Note also that STM32 is not an exception here: TI, ADI, Xilinx, Infineon, Microchip, Nuovoton do the same and some use of this could aslo be found on Nordic boards:
https://github.com/zephyrproject-rtos/zephyr/blob/a2f581f99459b3c9673f4403c346f9100952554a/boards/nordic/nrf7002dk/nrf7002dk_nrf5340_cpunet_defconfig#L13

Reply:
@nordicjm Feedback on my comment ?

Reply:
Please, open an issue to fix STM32 platform. All drivers using pinctrl should select the option. I guess it's ok to merge provided this gets fixed soon.

Reply:
> Please, open an issue to fix STM32 platform. All drivers using pinctrl should select the option. I guess it's ok to merge provided this gets fixed soon.

@gmarull Sure, working on it.
@nordicjm ^^ An example of an appropriate manner to request such change

Reply:
@erwango , @nordicjm , @gmarull  can we resolve this? It seems the only thing missing to move forward on merging this PR.

Reply:
Solved.

Reply:
Removed `CONFIG_PINCTRL`.